atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -943,10 +943,24 @@ protected boolean shouldDeleteAndRollback() {
     }
 
     @Override
+    final public void delete(TableReference tableRef, Set<Cell> cells) {
+        put(tableRef, Cells.constantValueMap(cells, PtBytes.EMPTY_BYTE_ARRAY), Cell.INVALID_TTL, Cell.INVALID_TTL_TYPE);
+    }
+
+    @Override
     public void put(TableReference tableRef, Map<Cell, byte[]> values) {
+        ensureNoEmptyValues(values);
         put(tableRef, values, Cell.INVALID_TTL, Cell.INVALID_TTL_TYPE);
     }
 
+    private void ensureNoEmptyValues(Map<Cell, byte[]> values) {
+        for (Entry<Cell, byte[]> cellEntry : values.entrySet()) {
+            if (cellEntry.getValue().length == 0) {
+                throw new IllegalArgumentException("AtlasDB does not currently support inserting empty (zero-byte) values.");
+            }
+        }
+    }
+
     public void put(TableReference tableRef, Map<Cell, byte[]> values, long ttlDuration, TimeUnit ttlUnit) {
```

Comment:
Good point. This method isn't part of the API and as far as I can tell should not be public - changed to private.